bokeh/models/ranges.py

```diff
@@ -90,6 +100,19 @@ class DataRange1d(DataRange):
     automatically computed end value.
     """)
 
+    bound_lower = Float(help="""
+    The minimum value that the range is allowed to go to - typically used to prevent
+    the user from panning/zooming/etc away from the data. If not provided, will
+    default to the min of the data.
+    """)
```

Comment:
Because `start` always has to be *something*, regardless of whether I supply that something, or whether that something gets computed. There is no case it which a start value literally not existing makes sense. Now, before there was only start, and making `start=None` as a way of saying "hey I'm not giving you a start value, figure it out yourself" seemed perfectly reasonable. In this context `None` is being used in the fairly common manner of `foo=None` keywoard arg that is computed if the user omits supplying it.

`lower_bound` by contrast has more "states", and we have to have a way to express all of them: 
* explicit user bounds 
* autocomputed bounds
* unbounded. 

Explicit is when a value is set, I think "auto" makes sense for "autocompute", that leaves `None`  for "unbounded", i.e. no bound, which also seems fairly sensible to me. Of course in this case `None` really means "None", not just a placeholder for a missing default. 

`start` and `lower_bound` sort of both "look like numbers" but they are different things, it's worth considering. `start` is a number, computed or given. `lower_bound` is a *policy* that describes how to do the computing for whenever `start` is computed. 

BTW we should document the precedence. What happens if a user sets `lower_bound` higher than the data min, who wins?